examples/wallet/app/components/RestoreWalletFromMnemonic.js

```diff
@@ -0,0 +1,176 @@
+// @flow
+import React, { useState } from 'react';
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+import Container from 'react-bootstrap/Container';
+import Row from 'react-bootstrap/Row';
+import typeof { setAccountFromMnemonic as SetAccountFromMnemonic } from '../actions/account';
+import { isValidMnemonic } from '../utils/mnemonic';
+
+type Props = {
+  setAccountFromMnemonic: SetAccountFromMnemonic
+};
+
+// FIXME: this has no error handling, neither while parsing the address
+// nor when fetching the balance.
+export default ({ setAccountFromMnemonic }: Props) => {
+  const checkIsValidMnemonicPhrase = function checkIsValidMnemonicPhrase() {
+    setIsMnemonicValid(isValidMnemonic(newMnemonicPhrase));
+  };
+
+  const handleSubmitMnemonic = function handleSubmitMnemonic(event) {
+    event.preventDefault();
+    if (mustCreateSpendingPassword) {
+      if (
+        isValidMnemonic(newMnemonicPhrase) &&
+        checkValidSpendingPassword(password, confirmPassword)
+      ) {
+        return Promise.all([
+          setAccountFromMnemonic(
+            newMnemonicPhrase,
+            newMnemonicPassword,
+            password
+          )
+        ]);
+      }
+    }
+    if (isValidMnemonic(newMnemonicPhrase)) {
+      return Promise.all([
+        setAccountFromMnemonic(newMnemonicPhrase, newMnemonicPassword, '')
+      ]);
+    }
+
+    setIsMnemonicValid(false);
+  };
+
+  const handleCheckCreateSpendingPassword = function handleCheckCreateSpendingPassword(
+    evt
+  ) {
+    setMustCreateSpendingPassword(evt.target.checked);
+    setHiddenSpendingPassword(!evt.target.checked);
+  };
+
+  const [isMnemonicValid, setIsMnemonicValid] = useState(true);
+  const [hiddenSpendingPassword, setHiddenSpendingPassword] = useState(false);
+  const [mustCreateSpendingPassword, setMustCreateSpendingPassword] = useState(
+    true
+  );
+
+  const [newMnemonicPhrase, setNewMnemonicPhrase] = useState('');
+
+  const [newMnemonicPassword, setNewMnemonicPassword] = useState('');
+
+  const checkValidSpendingPassword = function checkValidSpendingPassword(
```

Comment:
this is duplicated code from the other restoreWallet component

Reply:
deleted